Misc fixes for gwcli usability

mikechristie opened the pull request:
- Replace ring_buffer_size with max_data_area_mb in the disk create help command.

- Add a warning about all ACLs needing to be the same type in the auth help command.

- Document the export command.

- Replace the per client nochap command with a target wide one. For compat reasons the patch that removes the per client nochap support is in a separate patch and can be dropped if we feel we need to support it to avoid regressions with users that are using it in 2.7.

dillaman commented on gwcli/client.py:
```diff
@@ -283,7 +283,9 @@ def ui_command_auth(self, chap=None):
         Specifying 'nochap' will remove chap authentication for the client
-        across all gateways.
+        across all gateways. If there are multiple clients, CHAP must be
+        enabled for all clients or  disabled for all clients. gwcli does not
```
Nit: extra space between ```or``` and ```disabled```

dillaman commented on gwcli.8:
```diff
@@ -123,6 +123,12 @@ The IQN provided will be validated and defined to the configuration object. Addi
+.nf
+\fB/iscsi-target/<iqn>/ auth chap=<user>/<pswd> | nochap\fR
+.fi
+CHAP authentication can be reset target wide for all setup ACLs using the \fBchap=\fR parameter. The username and password defined here must then be used within the client'd login credentials for this iscsi target. To remove chap authentication for all setup ACLs use the \fBnochap\fR keyword. If there are multiple clients, CHAP must be enabled for all clients or disabled for all clients. gwcli do ...
```
Nit: ```client'd``` -> ```clients'```

dillaman commented on gwcli/client.py:
```diff
@@ -335,6 +335,18 @@ def ui_command_auth(self, chap=None):
+        if chap == 'nochap' or chap == '':
+            self.logger.error("CHAP must be disabled for all clients at the "
+                              "same time. Run the 'auth nochap' command from "
+                              "the target's node.")
```
Nit: the "node" terminology is a bit confusing to me. Perhaps is can be clarified by stating something like "target's node within gwcli"?

dillaman commented on gwcli.8:
```diff
@@ -163,9 +163,9 @@ The create command will define the client IQN to all gateways within the configu
 .TP
 .nf
-\fB/iscsi-target/<iqn>/hosts/<client_iqn>/ auth chap=<user>/<pswd> | nochap\fR
+\fB/iscsi-target/<iqn>/hosts/<client_iqn>/ auth chap=<user>/<pswd>\fR
```
Is this a limitation of the kernel or gwcli / ceph-iscsi-config? If it's just a limitation in the tooling, perhaps we can just highlight the inconsistent auth issue within the UI (like how it currently shows auth in red if not configured)?

mikechristie replied:
It's a limitation in the kernel.

mikechristie replied:
Oh yeah, but I think we can also detect and highlight the issue similar to when auth is not configured. I will add that in when I repush.

mikechristie commented:
Hey @dillaman I did a repush with your comments handled more or less.

I say the latter because for the last patch I changed the error message, but I also moved the command to the hosts node instead of the target. I think this might cause some confusion because I added the command to the "hosts" node which is the parent of all hosts, but I can imagine a sleepy admin might see the error message and think that it says the "host's" node.

dillaman commented on gwcli/client.py:
```diff
@@ -176,8 +176,47 @@ def delete(self, child):
+        auth nochap
+
+        """
+
```
Nit: validate that ```chap``` is None or equal to "nochap"

dillaman commented on gwcli/client.py:
```diff
@@ -330,6 +344,48 @@ def ui_command_auth(self, chap=None):
+                              "chap=<user>/<password>")
+            return
+
+        if chap == 'nochap' or chap == '':
```
Nit: can't hit ``` or chap == ''``` since it will hit ```if not chap``` above

mikechristie commented:
@dillaman 

Fixed both issues.

Also added 2 more help or auth related patches.

1. https://github.com/ceph/ceph-iscsi-cli/pull/113/commits/1993a414a7df108d18a7fc0c9b2d0b01f243f389

Use consistent name for image id in help commands.

2. https://github.com/ceph/ceph-iscsi-cli/pull/113/commits/d89ffb2e2b7363d2012aa353a3d05c41c50f8a85

Fix info output for a host's auth value so it matches the summary string values.

dillaman commented on gwcli/storage.py:
```diff
@@ -384,16 +384,16 @@ def ui_command_reconfigure(self, image_id, attribute, value):
         default.
 
-        image_id  : disk name (pool.image format)
+        image_id  : disk name (pool.image_id format)
```
Nit: it's still ```image``` here (I'll fix during merge)

dillaman commented:
Manually merged
